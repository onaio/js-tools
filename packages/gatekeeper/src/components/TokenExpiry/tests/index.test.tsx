import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Router } from 'react-router';
import { TokenExpired } from '../';

const history = createBrowserHistory();

describe('/components/errors/Fallback', () => {
  it('renders without crasshing', () => {
    shallow(<TokenExpired logoutUrl="/logout" />);
  });

  it('render correctly', () => {
    const wrapper = mount(
      <Router history={history}>
        <TokenExpired logoutUrl="/logout" />
      </Router>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('h2').text()).toEqual(
      'Your session has expired. Please click the link below to log in again.'
    );
    expect(wrapper.find('Link').text()).toEqual('Renew session');

    expect(wrapper.find('a').length).toMatchInlineSnapshot(`1`);
  });
});
