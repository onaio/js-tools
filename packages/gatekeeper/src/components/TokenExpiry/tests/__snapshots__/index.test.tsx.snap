// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/errors/Fallback render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <TokenExpired
    logoutLinkText="Renew session"
    logoutUrl="/logout"
    sessionExpiryText="Your session has expired. Please click the link below to start a new session"
  >
    <Container
      fluid={true}
      style={
        Object {
          "minHeight": "100vh",
          "position": "relative",
        }
      }
      tag="div"
    >
      <div
        className="container-fluid"
        style={
          Object {
            "minHeight": "100vh",
            "position": "relative",
          }
        }
      >
        <Row
          className="text-center"
          style={
            Object {
              "marginTop": "40px",
            }
          }
          tag="div"
        >
          <div
            className="text-center row"
            style={
              Object {
                "marginTop": "40px",
              }
            }
          >
            <Col
              md={
                Object {
                  "offset": 2,
                  "size": 8,
                }
              }
              sm="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <div
                className="col-sm-12 col-md-8 offset-md-2"
              >
                <h2>
                  Your session has expired. Please click the link below to start a new session
                </h2>
                <Link
                  to="/logout"
                >
                  <LinkAnchor
                    href="/logout"
                    navigate={[Function]}
                  >
                    <a
                      href="/logout"
                      onClick={[Function]}
                    >
                      Renew session
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Container>
  </TokenExpired>
</Router>
`;
